doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,452 @@
+.. role:: html(raw)
+   :format: html
+
+.. _mid_circuit_measurements:
+
+Mid-circuit measurements
+========================
```

Comment:
How about making the title "Dynamic circuits"?

Reply:
If this change is made, then maybe the "Available features" section should be renamed to "Mid-circuit measurements"? Let me know what you think. "Available features" is definitely more descriptive, so I don't have a preference.

Reply:
I initially thought the same, but then we have "conditional operators" under "mid-circuit measurements", which feels suboptimal.

@trbromley I agree that Dynamic circuits is the better advertising title. However, "Mid-circuit measurements" is technically the more accurate description of the page's content, I'd say. I'll go for "Dynamic quantum circuits" for now.

Reply:
I added a sub-header to the initial part that introduces `qml.measure` itself.
Overall, modified the header structure to

**Dynamic quantum circuits**
- Available features
  - Mid-circuit measurements
  - Resetting qubits
  - Postselecting mid-circuit measurements
  - Conditional Operators
  - Mid-circuit measurements statistics
- Simulation techniques
  - Deferred measurements
  - Dynamic one-shot sampling
  - Tree-traversal algorithm
  - Configuring mid-circuit measurements

Reply:
Nice! Yes I was thinking that while "mid-circuit measurements" is maybe more exact now, we can build on this "dynamic circuits" quickstart in future work.